src/puppetlabs/trapperkeeper/core.clj

```diff
@@ -93,6 +94,18 @@
    :post [(satisfies? app/TrapperkeeperApp %)]}
   (boot-services-with-config-fn services (constantly config-data)))
 
+(schema/defn boot-with-config-fn :- (schema/protocol app/TrapperkeeperApp)
```

Comment:
Wondering if we might be able to convince ourselves that the config doesn't need to be a function?  (We might not, if it was intended to allow for a dynamic config that could change on say SIGHUP, but I suspect that's  not how you'd ever want to coordinate such a thing.)  And if we can, then perhaps this could just be boot-with-config.